airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadStateManager.java

```diff
@@ -41,7 +41,13 @@ public void updatePrimaryKeyLoadState(final AirbyteStreamNameNamespacePair pair,
 
   // Returns the previous state emitted, represented as a {@link PrimaryKeyLoadStatus} associated with
   // the stream.
-  public abstract PrimaryKeyLoadStatus getPrimaryKeyLoadStatus(final AirbyteStreamNameNamespacePair pair);
+  public PrimaryKeyLoadStatus getPrimaryKeyLoadStatus(final AirbyteStreamNameNamespacePair pair) {
+    if (pairToPrimaryKeyLoadStatus.containsKey(pair)) {
```

Comment:
simple doing `get()` does the same thing, no?

Reply:
yeah you are right... I was paranoid that this was null

Reply:
fixed!